server/rules-engine/index.js

```diff
@@ -0,0 +1,172 @@
+const DEBUG = false
+const Engine = require('json-rules-engine').Engine
+const enums = require('./enums')
+const Rule = require('json-rules-engine').Rule
+const operators = require('./operators')
+const factHandler = require('./factHandler')
+
+const _buildStandardRules = function (conditions, ruleName, eventType) {
+  if (DEBUG) console.log('Setting up standard rules with conditions ', conditions)
+  return conditions.map(condition => new Rule({
+    conditions: {
+      all: [condition]
+    },
+    event: {
+      type: eventType
+    },
+    name: ruleName,
+    priority: 5
+  }))
+}
+
+const _setupStandardRule = function (conditions, ruleName, eventType) {
+  const rules = _buildStandardRules(conditions, ruleName, eventType)
+  for (const rule of rules) {
+    this.engine.addRule(rule)
+  }
+  this.engine.on('success', function (event, almanac, ruleResult) {
+    if (event.type === eventType) {
+      almanac.addRuntimeFact(enums.ruleRejected, true)
```

Comment:
This code seems counter intuitive. I'm not clear why are in a position where we need to record the fact that a rule failed in the success event. I haven't seen this pattern in any of the examples I've looked at.